packages/terra-show-hide/src/ShowHide.jsx

```diff
@@ -0,0 +1,158 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Toggle from 'terra-toggle';
+import { injectIntl, intlShape } from 'terra-base';
+import IconChevronDown from 'terra-icon/lib/icon/IconChevronDown';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import styles from './ShowHide.module.scss';
+import Button from './_ShowHideButton';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Content in the body of the component that will be shown or hidden
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * Text that will be visible to the user while the component is collapsed
+   */
+  preview: PropTypes.string.isRequired,
+  /**
+   * Button text that will be displayed when the component is collapsed
+   */
+  collapsedButtonText: PropTypes.string,
+  /**
+   * Button text that will be displayed when the component is expanded
+   */
+  expandedButtonText: PropTypes.string,
+  /**
+   * Icon displayed next to the button text
+   */
+  icon: PropTypes.element,
+  /**
+   * The intl object to be injected for translations. Provided by the injectIntl function.
+   */
+  intl: intlShape.isRequired,
+  /**
+   * Sets the animation for the component when it is expanded or collapsed
+   */
+  isAnimated: PropTypes.bool,
+  /**
+   * Sets the initial state of the component to open and expanded
+   */
+  isInitiallyOpen: PropTypes.bool,
+  /**
+   * Callback function triggered when the component is collapsed
+   */
+  onClose: PropTypes.func,
+  /**
+   * Callback function triggered when the component is expanded
+   */
+  onOpen: PropTypes.func,
+};
+
+const defaultProps = {
+  icon: <IconChevronDown />,
+  isAnimated: false,
+  isInitiallyOpen: false,
+};
+
+const animatedStyle = {
+  // fix for the toggle oscillation when setting isAnimated to true
+  transitionDelay: '0.3s',
+  WebkitTransitionDelay: '0.3s', /* Safari */
+};
+
+class ShowHide extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isOpen: props.isInitiallyOpen,
+    };
+
+    this.handleOnClick = this.handleOnClick.bind(this);
+  }
+
+  handleOnClick(e) {
+    e.preventDefault();
```

Comment:
What default action do we need prevented?

Reply:
I copied this function from terra-toggle-button. I assumed it was a bug fix for something. It might not be necessary now though. @bjankord do you recall why this line was added to terra-toggle-button?

Reply:
I don't, might test removing it and see if things still work as intended.

Reply:
Alright I tested it and it can be removed